app/src/main/java/org/onionshare/android/tor/TorManager.kt

```diff
@@ -219,6 +263,26 @@ class TorManager @Inject constructor(
         TorControlConnection(inputStream, outputStream)
     }
 
+    private fun changeStartingState(progress: Int, onion: String? = null) {
+        val oldStartingState = state.value as? TorState.Starting
+        if (oldStartingState == null) LOG.warn("Old state was not Starting, but ${state.value}")
```

Comment:
No, because we do `_state.value = newState ?: TorState.Starting(` which uses a new `Starting` state if `newState` is `null`. That is what this elvis operator does: Assign `newState` except when it is `null`, then use what is on the right side of the operator.

Reply:
One could also consider to return without setting a new state as we are obviously somewhere unexpected. An assertion error might be a bit harsh without a crash catcher.